photon-core/src/main/java/org/photonvision/vision/pipe/impl/Calibrate3dPipe.java

```diff
@@ -222,13 +228,36 @@ protected CameraCalibrationCoefficients calibrateMrcal(
         JsonMatOfDouble distortionCoefficientsMat =
                 new JsonMatOfDouble(1, 8, CvType.CV_64FC1, Arrays.copyOfRange(result.intrinsics, 4, 12));
 
-        // Calculate optimized board poses manually. We get this for free from mrcal too, but that's not
+        // Calculate optimized board poses manually. We get this for free from mrcal
+        // too, but that's not
         // JNIed (yet)
         List<Mat> rvecs = new ArrayList<>();
         List<Mat> tvecs = new ArrayList<>();
+
         for (var o : in) {
             var rvec = new Mat();
             var tvec = new Mat();
+
+            Point3[] oPoints = o.objectPoints.toArray();
+            Point[] iPoints = o.imagePoints.toArray();
+
+            List<Point3> outputOPoints = new ArrayList<Point3>();
+            List<Point> outputIPoints = new ArrayList<Point>();
+
+            for (int i = 0; i < iPoints.length; i++) {
+                if (iPoints[i].x != -1 && iPoints[i].y != -1) {
```

Comment:
Is comparing floats to ints
```suggestion
                if (iPoints[i].x != -1 && iPoints[i].y != -1) {
```
 here OK? Also generally confused what this code is doing. Please comment